### PR TITLE
Customer change email

### PR DESCRIPTION
**Pull request by gillesbourgeat**
Hello,

Add the ability for a user to change their email.
Activation is manageable from the customer_change_email config

### CODE REVIEW

File: setup/insert.sql [hunk 1]
```diff
@@ -147,7 +150,8 @@ INSERT INTO `config_i18n` (`id`, `locale`, `title`, `description`, `chapo`, `pos
 (33, 'fr_FR', 'Utiliser un cookie persistant pour memoriser le panier du client', NULL, NULL, NULL),
 (34, 'fr_FR', 'Nom du cookie de stockage du panier', NULL, NULL, NULL),
 (35, 'fr_FR', 'Durée de vie du cookie du panier dans le navigateur du client, en secondes', NULL, NULL, NULL),
-(36, 'fr_FR', 'Durée de vie du cookie de la session dans le navigateur du client, en secondes', NULL, NULL, NULL)
+(36, 'fr_FR', 'Durée de vie du cookie de la session dans le navigateur du client, en secondes', NULL, NULL, NULL),
+(36, 'fr_FR', 'Permettre aux utilisateurs de changer leurs email. 1 pour oui, 0 pour non', NULL, NULL, NULL)
```
Comment (lunika) [inline setup/insert.sql]: this is not the good id, the good id is 37

File: local/modules/Front/Controller/CustomerController.php [hunk 2]
```diff
@@ -331,8 +332,15 @@ public function updateAction()
 
                 $customerChangeEvent = $this->createEventInstance($form->getData());
                 $customerChangeEvent->setCustomer($customer);
-                // We do not allow customer email modification
-                $customerChangeEvent->setEmailUpdateAllowed(false);
+
+                $configCustomerChangeEmail = intval(ConfigQuery::read('customer_change_email', 0));
+
+                if ($configCustomerChangeEmail) {
```
Comment (lunika) [inline local/modules/Front/Controller/CustomerController.php]: you can directly test the value returned by ConfigQuery::read here, it is not needed to put it in an intermediary variable

File: setup/update/sql/2.2.0-alpha1.sql [hunk 3]
```diff
@@ -24,4 +24,14 @@ INSERT INTO  `hook_i18n` (`id`, `locale`, `title`, `description`, `chapo`) VALUE
 
 ALTER TABLE `admin_log` ADD `resource_id` INTEGER AFTER `resource` ;
 
+INSERT INTO `config` (`name`, `value`, `secured`, `hidden`, `created_at`, `updated_at`) VALUES
```
Comment (lunika) [inline setup/update/sql/2.2.0-alpha1.sql]: look in previous sql files how we did the addition of new variable config. You have to retrieve first the last id an the insert in the table the new records.

**Comment by gillesbourgeat** [comment]
in a future pull request, add verfication email for create and update customer

**Comment by lovenunu** [comment]
OMG @gillesbourgeat you're so opensource ! :D

**Comment by lunika** [comment]
Thanks,

please complete the readme and explain how to use it.

**Pull request closed without merging**